pages/index.js

```diff
@@ -32,11 +38,12 @@ export default function Home() {
         <h3>Name my pet</h3>
         <form onSubmit={onSubmit}>
           <input
+            className={!validInput && styles.invalidInput}
             type="text"
             name="animal"
             placeholder="Enter an animal"
             value={animalInput}
-            onChange={(e) => setAnimalInput(e.target.value)}
+            onChange={onChangeHandler}
```

Comment:
Have you considered using the [`minlength` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/minlength)?

Reply:
I've added it now, can you take a look?

Reply:
LGTM